src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AddHandler.scala

```diff
@@ -33,85 +34,7 @@ class AddHandler(authorizationProvider: AuthorizationProvider) extends Actor {
-    if(authorizationProvider.isAuthorizedForCreation(agoraEntity, username)) {
-      val method = agoraBusiness.insert(agoraEntity.copy(createDate = Option(new DateTime())), username)
-      authorizationProvider.createEntityAuthorizations(method, username)
-      context.parent ! RequestComplete(Created, method)
-    }
-    else {
-      context.parent ! RequestComplete(BadRequest, AgoraError("You don't have permission to create items in " + agoraEntity.namespace))
-    }
-  }
-
-}
-
-case class MethodImportResolver(username: String, business: AgoraBusiness, authorizationProvider: AuthorizationProvider) {
-
-  def importResolver(importUri: String): WdlSource = {
-    ImportResolverHelper.validateUri(importUri)
-    val method = ImportResolverHelper.resolve(importUri, business, username)
-    if (method == None) throw new SyntaxError(s"Can't resolve import: $importUri")
-    val canRead = authorizationProvider.isAuthorizedForRead(method.get, username)
-    if (canRead) method.get.payload.get
-    else throw new SyntaxError(s"Can't resolve import: $importUri")
-  }
-}
-
-object ImportResolverHelper {
-
-  private val methodScheme = "methods://"
-
-  def resolve(uri: String, business: AgoraBusiness, username: String): Option[AgoraEntity] = {
-    val withoutScheme = removeScheme(uri)
-    val parts = uriParts(withoutScheme)
-    val namespace = parts(0)
-    val name = parts(1)
-    val snapshotId = parts(2).toInt
-    val queryMethod = AgoraEntity(namespace = Option(namespace), name = Option(name), snapshotId = Option(snapshotId))
-    val method = business.findSingle(queryMethod.namespace.get,
-                                    queryMethod.name.get,
-                                    queryMethod.snapshotId.get,
-                                    Seq(AgoraEntityType.Task, AgoraEntityType.Workflow),
-                                    username)
-    method
-  }
-
-  def validateUri(importUri: String) = {
-    if (!importUri.startsWith(methodScheme)) {
-      throw new SyntaxError(s"Unsupported import uri: $importUri. Must start with $methodScheme.")
-    }
-    val withoutScheme = removeScheme(importUri)
-    val parts = uriParts(withoutScheme)
-    if (parts.size != 3) {
-      throw new SyntaxError(s"Malformed methods uri. Must contain three parts, separated by '.': $importUri")
-    }
-    if (!isIdInt(parts(2))) {
-      throw new SyntaxError(s"Malformed methods uri. SnapshotId must be integer: $importUri")
-    }
-  }
-
-  private def removeScheme(uri: String): String = {
-    uri.substring(methodScheme.length)
-  }
-
-  private def isIdInt(idStr: String): Boolean = {
-    idStr.forall(_.isDigit)
-  }
-
-  private def uriParts(uriWithoutScheme: String): Array[String] = {
-    uriWithoutScheme.split("\\.")
+    context.parent ! RequestComplete(Created, agoraBusiness.insert(agoraEntity.copy(createDate = Option(new DateTime())), username))
```

Comment:
AgoraBusiness.insert performs the entity copy to add createDate. See
https://github.com/broadinstitute/agora/pull/84/files#diff-1768f9d51119592601a48548da8f9904R16

Remove the copy here